### PR TITLE
[kokoro] Added an apidiff bot to our CI.

### PR DESCRIPTION
**Pull request by jverkoey**
Adds support for generating API diffs for PRs.

## How does it work?

There is a new job that runs alongside our PRs, `kokoro/pr/apidiff`. This job will run the `.kokoro` script with the `apidiff` environment. In this environment, we run our `scripts/release` API diff command and post the output as a comment to our pull request, if any API diff exists.

The intent of this bot is to allow us to quickly get an understanding of any API changes in a given PR. Most notably, if an API is being deleted without first being deprecated, this bot will give us a good opportunity to catch that before it lands in our release.

## Example output

screen shot 2018-05-09 at 9 49 11 am

### CODE REVIEW

**Comment by romoore** [comment]
Hello, bot!

**Comment by romoore** [comment]
🐦  **jverkoey** deleted a comment from **material-prbot** 30 seconds ago

**Comment by jverkoey** [comment]
screen shot 2018-05-10 at 2 12 20 pm

We have a winner!

**Review by romoore — COMMENTED** [review]
2371 lines changed?!

**Comment by jverkoey** [comment]
It's all of the github comment handling.

**Comment by jverkoey** [comment]
And more. I didn't trim the impl down from the issue sync. But perhaps I should...

File: .kokoro [hunk 1]
```diff
@@ -230,6 +230,97 @@ generate_website() {
   fi
 }
 
+generate_apidiff() {
```
Comment (romoore) [inline .kokoro]: Maybe docs?

File: .kokoro [hunk 2]
```diff
@@ -230,6 +230,97 @@ generate_website() {
...
+    # Move into our cloned repo
+    cd github/repo
+
+    # Install sourcekitten, a dependency of the apidiff tool
+    brew install sourcekitten
+  fi
+
+  if [ -z "$GITHUB_API_TOKEN" ]; then
+    echo "GITHUB_API_TOKEN must be set to a github token with public_repo scope."
+    exit 1
```
Comment (romoore) [inline .kokoro]: Consider (a follow-up change with) a `usage` function.

**Comment by jverkoey** [comment]
Removed the unused APIs.

**Comment by jverkoey** [comment]
Just 638 now!

File: .kokoro (hunk 1 above)
Comment (jverkoey) [reply .kokoro]: Done.

File: .kokoro (hunk 2 above)
Comment (jverkoey) [reply .kokoro]: Done.

**Comment by jverkoey** [comment]
Gentle ping. Needs review.

File: .kokoro [hunk 3]
```diff
@@ -230,6 +230,125 @@ generate_website() {
...
+  if [ -z "$KOKORO_GITHUB_PULL_REQUEST_NUMBER" ]; then
+    echo "KOKORO_GITHUB_PULL_REQUEST_NUMBER must be set to a github pull request number."
+    usage
+    exit 1
+  fi
+
+  BUILDS_TMP_PATH=$(mktemp -d)
+  SEEN_XCODES_FILE_PATH="$BUILDS_TMP_PATH/seen_xcodes"
+  touch "$SEEN_XCODES_FILE_PATH"
+  xcodes=$(ls /Applications/ | grep "Xcode")
```
Comment (romoore) [inline .kokoro:283]: (Nit:) Is there a semantic difference between `lowercase_variables` and `UPPERCASE_VARIABLES`?

File: scripts/github-comment/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+.build
+*.xcodeproj
```
Comment (romoore) [inline scripts/github-comment/.gitignore:2]: Nit/Formatting: Missing ending newline?

**Review by romoore — COMMENTED** [review]
I hate to be picky, but can this be pushed up as 2 or 3 changes?

1. Swift utility (github-comment?)
2. Bash script to run the Swift utility
3. (Maybe as part of 2) Update the release script to include apidiff?

**Comment by jverkoey** [comment]
Breaking this up into separate PRs.

First: https://github.com/material-components/material-components-ios/pull/4169

**Pull request closed without merging**